packages/examples/src/diffie-hellman/diffie-hellman.ts

```diff
@@ -0,0 +1,158 @@
+import { Choreography, Located } from "@choreography-ts/core";
+import { ExpressBackend } from "@choreography-ts/backend-express";
+
+// Typescript implementation of the Diffie-Hellman key exchange algorithm
+// Ported from the HasChor implementation here: https://github.com/gshen42/HasChor/blob/42ae1ef9a500dadd82f0cfe5dee3c2aa631d8f4d/examples/diffiehellman/Main.hs
+
+//////////////////////
+// Helper functions //
+/////////////////////
+
+// Return all divisors of `x`
+const divisors = (x: number): number[] => {
+  let divs: number[] = [1, x];
+  for (let y = 2; y <= x / 2; y++) {
+    if (x % y == 0) divs.push(y);
+  }
+  return divs;
+};
+
+// Checks if input number is prime
+const isPrime = (x: number): boolean => {
+  let divs: number[] = divisors(x);
+  if (divs.length == 2) return true;
+  return false;
+};
+
+// Prime number generator
+const primeNums = function* () {
+  let y = 1;
+  while (y++ > 0) {
+    if (isPrime(y)) yield y;
+  }
+};
+
+// Return random integer in range [min, max]
+// https://stackoverflow.com/a/7228322
+const randInRange = (min: number, max: number) =>
+  Math.floor(Math.random() * (max - min + 1) + min);
+
+///////////////////////////////////
+// Diffie-Hellman implementation //
+///////////////////////////////////
+const locations = ["alice", "bob"];
+export type Location = (typeof locations)[number];
+
+export const diffieHellman = <A extends Location, B extends Location>(
+  a: A,
+  b: B
+) => {
+  const choreo: Choreography<
+    Location,
+    [Located<boolean, A>], // Specify boolean argument specifically for location "A" ("A|B" instead would mean for both)
+    [Located<number, A>, Located<number, B>]
+  > = async ({ locally, comm }, [arg]) => {
+    // The `arg` specifies whether `a` should wait to initiate key exchange
+
+    // Wait for alice to start key exchange (if desired)
+    await locally(a, (unwrap: any) => {
```

Comment:
```suggestion
    await locally(a, (unwrap) => {
```

The type of `unwrap` must be inferred so it cannot unwrap values located at wrong locations

Reply:
Ok. I actually did that because otherwise TS was giving me a bunch of errors regarding invalid arithmetic types otherwise wherever I used `mod`. Specifically the following error:
```
src/diffie-hellman/diffie-hellman.ts:113:45 - error TS2363: The right-hand side of an arithmetic operation must be of type 'any', 'number', 'bigint' or an enum type.

113       (unwrap) => unwrap(gb) ^ unwrap(sb) % unwrap(pb)
                                                ~~~~~~~~~~
```

Reply:
And I'm not aware of any other way around this issue...

Reply:
I think it has to do with the `void` type that I mentioned in the other comment?

Reply:
So `unwrap: number` instead of `unwrap: any`?

Reply:
well, it's `<T>(val: Located<T, B>) => T` to be exact. It should be inferred and you should not annotate here.

Reply:
I just made the changes and pushed them

Reply:
> well, it's `<T>(val: Located<T, B>) => T` to be exact. It should be inferred and you should not annotate here.

Oh, I see. I thought we should assume they're numbers here though since that's what we're working with, no? And don't we want to protect against bad types, like say if a string is given or something?